File: Assets/Android Studio Project~/UnityMobileModules/src/main/java/com/UnityMobileModules/Alert.java
package com.UnityMobileModules;

import com.unity3d.player.UnityPlayer;
import android.app.AlertDialog;
import android.content.DialogInterface;

public class Alert
{
    /**
     *  Shows a alert to the user
     * @param alertTitle - Title of the alert
     * @param alertBody - Text of the alert
     * @param cancelButtonText - Text to display on the cancel button
     */
    public static void ShowAlert(String alertTitle, String alertBody, String cancelButtonText)
    {
        AlertDialog.Builder builder = new AlertDialog.Builder(UnityPlayer.currentActivity);
        builder.setTitle(alertTitle)
                .setMessage(alertBody)
                .setCancelable(false)
                .setPositiveButton(cancelButtonText, new DialogInterface.OnClickListener() {
                    //by default doing nothing just closes the dialog
                    //this does nothing, it can do stuff though
                    public void onClick(DialogInterface dialog, int id) {}
                });
        AlertDialog alert = builder.create();
        alert.show();
    }
}

File: Assets/UnityMobileModules/Alert/Alert.cs
namespace UnityMobileModules
{
	/// <summary>
    /// Shows an Alert
    /// </summary>
    public static partial class Alert
    {
        /// <summary>
        /// Shows an Alert
        /// </summary>
        /// <param name="alertTitle">Title of the alert</param>
        /// <param name="alertBody">Text of the alert</param>
        /// <param name="cancelButtonText">Text to display on the cancel button</param>
        public static void ShowAlert(string alertTitle, string alertBody, string cancelButtonText)
        {
#if UNITY_EDITOR
            return;
#elif UNITY_ANDROID
            ShowAlert_Android(alertTitle, alertBody, cancelButtonText);
#elif UNITY_IOS

#endif
        }
    }
}
